kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/ProxyConfigConfigMap.java

```diff
@@ -94,16 +98,26 @@ public static List<VolumeMount> secureVolumeMounts(ManagedWorkflowAndDependentRe
         return mounts.stream().toList();
     }
 
+    @Override
+    public boolean isMet(DependentResource<ConfigMap, KafkaProxy> dependentResource, KafkaProxy primary, Context<KafkaProxy> context) {
+        try {
+            var configuration = generateProxyConfig(context);
+            context.managedWorkflowAndDependentResourceContext().put(CONFIGURATION_DATA_KEY, configuration);
+            return true;
+        }
+        catch (IllegalConfigurationException ice) {
+            context.managedWorkflowAndDependentResourceContext().put(CONFIGURATION_DATA_KEY, null);
+            return false;
+        }
+    }
+
     @Override
     protected ConfigMap desired(KafkaProxy primary,
                                 Context<KafkaProxy> context) {
-        Clock clock = KafkaProxyContext.proxyContext(context).clock();
-        var data = new ProxyConfigData();
-        data.setProxyConfiguration(generateProxyConfig(context));
-
-        ProxyModel proxyModel = KafkaProxyContext.proxyContext(context).model();
-        addResolvedRefsConditions(clock, proxyModel, data);
-        addAcceptedConditions(clock, proxyModel, data);
+        // the configuration object won't be present if isMet has returned false
+        // this is the case if the dependant resource is to be removed.
```

Comment:
does desired get called if the precondition is false? seems a bit surprising that it gets the desired state of the dependent resource if JOSDK is going to delete it.

Reply:
> does desired get called if the precondition is false? seems a bit surprising that it gets the desired state of the dependent resource if JOSDK is going to delete it.

In the delete case, desired is called in order to identify the object that needs to be deleted.